cesium/writer_stream.go

```diff
@@ -233,15 +233,22 @@ func (w *streamWriter) sendRes(req WriterRequest, ack bool, err error, end telem
 	}
 }
 
-func (w *streamWriter) write(req WriterRequest) (err error) {
+func (w *streamWriter) write(ctx context.Context, req WriterRequest) (err error) {
 	for _, idx := range w.internal {
-		req.Frame, err = idx.Write(req.Frame)
+		req.Frame, err = idx.Write(ctx, req.Frame)
 		if err != nil {
 			if errors.Is(err, control.Unauthorized) && !*w.ErrOnUnauthorized {
 				return nil
 			}
 			return err
 		}
+
+		if w.Mode.AutoCommit() {
+			_, err = idx.Commit(ctx, true)
+			if err != nil {
+				return
```

Comment:
is there no other action that needs to be taken on error?

Reply:
This is a behaviour I'm not sure about: ideally, we would want a failed commit (which is most likely due to the domain of the written telemetry being wrong) to also "revert" the write so the writer could re-try writing correct data. However, this is really hard and could be really slow since the data is already written to disk. So in this case we just return an error and not do anything. This is also partially because our driver when encountering a write error would just make a new writer instead of trying to write again.

Is this the desired behaviour? Does this make sense?

Reply:
@Lham42 Are you confused about why he's not returning the error here. If so, it's because go has implicit returns of named variables.

Reply:
Ok that makes sense. It wasn't clear to me if errors were handled but if they can be and are handled by the caller then this is fine.